### PR TITLE
Assorted fixes to making JSON the default

### PR DESCRIPTION
**Pull request by gsmet**
- Fix provider detection to avoid including all the providers if JSON is the default and there are no media types annotations (it's a 10 MB penalty for the native executables)
- Generalize this to the REST Client as it was somehow already working if you had a JSON server extension around
- Fix some typos in the documentation and extend it
- Take into account the JSON default in the Hibernate Validator JAX-RS interceptor.

### CODE REVIEW

File: docs/src/main/asciidoc/rest-client.adoc [hunk 1]
```diff
@@ -142,9 +141,15 @@ CDI injection as a REST Client
...
-it is heavily recommended to annotate your endpoints with them to define precisely the expected content-types.
+When a JSON extension is installed such as `quarkus-rest-client-jsonb` or `quarkus-rest-client-jackson`, Quarkus will use the `application/json` media type
+by default for most return values, unless the media type is explicitly set via
+`@Produces` or `@Consumes` annotations (there are some exceptions for well known types, such as `String` and `File`, which default to `text/plain` and `application/octet-stream`
+respectively).
+
+If you don't want JSON by default you can set `quarkus.resteasy-json.default-json=false` and the default will change back to being auto-negotiated. If you set this
+you will need to add `@Produces(MediaType.APPLICATION_JSON)` and `@Consumes(MediaType.APPLICATION_JSON)` to your endpoints in order to use JSON.
 
+If you don't rely on the JSON default, it is heavily recommended to annotate your endpoints with the `@Produces` and `@Consumes` annotations to define precisely the expected content-types.
```
Comment (geoand) [inline docs/src/main/asciidoc/rest-client.adoc:152]: :+1:

**Review by geoand — APPROVED** [review]

**Comment by gsmet** [comment]
I added an additional commit to take into account the default in the Hibernate Validator JAX-RS interceptor.

This default behavior change wasn't that simple after all :).

**Comment by gsmet** [comment]
I force-pushed some changes to fix CI.

File: extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java [hunk 2]
```diff
@@ -45,6 +46,10 @@ public void validateConstructorInvocation(InvocationContext ctx) throws Exceptio
         MediaType[] producedMediaTypes = MediaTypeHelper.getProduces(method.getDeclaringClass(), method);
 
         if (producedMediaTypes == null) {
+            if (resteasyConfigSupport.isJsonDefault()) {
+                return Arrays.asList(MediaType.APPLICATION_JSON_TYPE);
```
Comment (gastaldi) [inline extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java:50]:
```suggestion
                return Collections.singletonList(MediaType.APPLICATION_JSON_TYPE);
```

**Review by gastaldi — APPROVED** [review]

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**